hash.c

```diff
@@ -1300,8 +1300,11 @@ hash_ar_foreach_iter(st_data_t key, st_data_t value, st_data_t argp, int error)
 
     if (error) return ST_STOP;
 
+    ar_table *tbl = RHASH_AR_TABLE(arg->hash);
     int status = (*arg->func)((VALUE)key, (VALUE)value, arg->arg);
-    /* TODO: rehash check? rb_raise(rb_eRuntimeError, "rehash occurred during iteration"); */
+    if (!RHASH_AR_TABLE_P(arg->hash) || RHASH_AR_TABLE(arg->hash) != tbl) {
```

Comment:
I know this is copying from similar code using `RHASH_ST_TABLE`, but that check might actually never trigger. RHASH_ST_TABLE() just adds an offset to the argument, so for the same hash, it's always going to be the same. `arg->hash` is never going to be changed by the callback, so that check will always be false.

Reply:
I changed this condition to check if the bound increases. That should work, right?